scripts/install/calibreweb.sh

```diff
@@ -88,13 +88,36 @@ function _install_calibreweb() {
 }
 
 _install_kepubify() {
-    echo_progress_start "Installing kepubify"
-    wget -q "https://github.com/pgaskin/kepubify/releases/download/v3.1.2/kepubify-linux-64bit" -O /tmp/kepubify >> $log 2>&1
-    chmod a+x /tmp/kepubify
-    mv /tmp/kepubify /usr/local/bin/kepubify
-    #TODO and figure out if it's needed for all cases or not
-    echo_progress_done
-}
+	echo_progress_start "Installing kepubify"
+
+	case "$(_os_arch)" in
+		"amd64")
+			kepubify_arch="64bit"
+			;;
+		"arm64")
+			kepubify_arch="arm64"
+			;;
+		"armhf")
+			kepubify_arch="arm"
+			;;
+		*)
+			echo_error "$(_os_arch) not supported by this app"
+			exit 1
+			;;
+	esac
+
+	kepubify_url="https://github.com/pgaskin/kepubify/releases/latest/download/kepubify-linux-${kepubify_arch}"
+	kepubify_url_status="$(curl -sLo /dev/null -w "%{http_code}" "${kepubify_url}")"
+
+	if [[ "${kepubify_url_status}" -eq '200' ]]; then
+		wget "${kepubify_url}" -O /usr/local/bin/kepubify
+		chmod 755 /usr/local/bin/kepubify
+	else
+		echo_error "kepubify github release URL status: ${kepubify_url_status}"
+		exit 1
+	fi
+	echo_progress_done
+} 2>> "${log}"
```

Comment:
The reason this will work is because the release [artifacts do not include version info](https://github.com/pgaskin/kepubify/releases) so this url will always provide the latest release asset for the relevant arch.

https://github.com/pgaskin/kepubify/releases/latest/download/kepubify-linux-64bit
https://github.com/pgaskin/kepubify/releases/latest/download/kepubify-linux-arm64
https://github.com/pgaskin/kepubify/releases/latest/download/kepubify-linux-arm

We are ignoring 32bit

So using the case statement we can create a url to always:

- pull the latest release of kepubify
- always get the matching arch for supported systems

So this fixes a few things

- This fixes the issue of having an old version hardlinked
- It fixes the single arch `64bit` configuration by supporting arm64 and arm
- Check the URL is valid for a `200` return code.
- errors out if the url fails.

I think this fixes the main problem and well as future proofs it.

Reply:
makes sense to me!